src/openvic-simulation/modifier/StaticModifierCache.cpp

```diff
@@ -1,115 +1,166 @@
+			key_map, modifier.get_identifier(), ONE_EXACTLY,
+			expect_dictionary(modifier_manager.expect_base_country_modifier(modifier))
+		);
+	};
 
-		if (mutable_modifier != nullptr) {
-			if (multiplier != fixed_point_t::_1()) {
-				*mutable_modifier *= multiplier;
+	const auto set_country_event_modifier = [&key_map, &modifier_manager, &ret](
+		Modifier const*& modifier_ptr, std::string_view const&& identifier, IconModifier::icon_t const& icon
+	) -> void {
+		ret &= add_key_map_entry(
+			key_map, identifier, ONE_EXACTLY,
+			[&modifier_manager, &modifier_ptr, &identifier, &icon]
+			(ast::NodeCPtr const& value)->bool {
+				modifier_ptr = modifier_manager.get_event_modifier_by_identifier(identifier);
+				if(modifier_ptr != nullptr) {
+					return true; //an event modifier overrides a static modifier with the same identifier.
+				}
+
+				ModifierValue modifier_value {};
+				bool has_parsed_modifier = expect_dictionary(modifier_manager.expect_base_country_modifier(modifier_value))(value);
+				has_parsed_modifier &= modifier_manager.add_event_modifier(identifier, std::move(modifier_value), icon, STATIC);
+				if(has_parsed_modifier) {
+					modifier_ptr = modifier_manager.get_event_modifier_by_identifier(identifier);
+				}
+				return has_parsed_modifier;
 			}
+		);
+		
+	};
 
-			modifier = mutable_modifier;
-		} else {
-			Logger::error("Failed to set static modifier: ", name);
-			ret = false;
-		}
+	const auto set_static_province_modifier = [&key_map, &modifier_manager, &ret](
+		Modifier& modifier
+	) -> void {
+		ret &= add_key_map_entry(
+			key_map, modifier.get_identifier(), ONE_EXACTLY,
+			expect_dictionary(modifier_manager.expect_base_province_modifier(modifier))
+		);
 	};
 
 	// Country modifiers
-	set_static_modifier(very_easy_player, "very_easy_player");
-	set_static_modifier(easy_player, "easy_player");
-	set_static_modifier(hard_player, "hard_player");
-	set_static_modifier(very_hard_player, "very_hard_player");
-	set_static_modifier(very_easy_ai, "very_easy_ai");
-	set_static_modifier(easy_ai, "easy_ai");
-	set_static_modifier(hard_ai, "hard_ai");
-	set_static_modifier(very_hard_ai, "very_hard_ai");
+	set_static_country_modifier(very_easy_player);
+	set_static_country_modifier(easy_player);
+	set_static_country_modifier(hard_player);
+	set_static_country_modifier(very_hard_player);
+	set_static_country_modifier(very_easy_ai);
+	set_static_country_modifier(easy_ai);
+	set_static_country_modifier(hard_ai);
+	set_static_country_modifier(very_hard_ai);
+
+	ret &= add_key_map_entry(
+		key_map, base_modifier.get_identifier(), ONE_EXACTLY,
+		expect_dictionary_keys_and_default(
+			modifier_manager.expect_base_country_modifier(base_modifier),
+			"supply_limit", ZERO_OR_ONE, [this, &modifier_manager](auto const& value) -> bool {
+				return modifier_manager._add_modifier_cb(
+					base_modifier,
+					modifier_manager.get_modifier_effect_cache().get_supply_limit_global_base(),
+					value
+				);
+			}
```

Comment:
Why the special handling? Is this effect valid in the same way across multiple modifier types? Also use `ast::NodeCPtr` instead of `auto const&`.

Reply:
Ofcourse this is yet another special case, we all know paradox loves those.
This is the only context in which you can add to the global base supply limit.
In other contexts you either add to the local supply limit or modify the global supply limit.